<launch>
    <!-- ObjectDetection -->
    <arg name="SOURCE" default="/zed2/zed_node/rgb/image_rect_color" /> 
    <arg name="DEPTH_INPUT" default="/zed2/zed_node/depth/depth_registered" />
    <arg name="CAMERA_INFO" default="/zed2/zed_node/depth/camera_info" />
    <arg name="ROS_INPUT" default="True" />
    <arg name="USE_ACTIVE_FLAG" default="False" />
    <arg name="VERBOSE" default="False" />
    <arg name="DEPTH_ACTIVE" default="True" />
    <arg name="USE_YOLO8" default="False" />
    <arg name="YOLO_MODEL_PATH" default="[yolov8n.pt]" />
    <arg name="YOLO5_MODELS_PATH" default="[]" />
    <arg name="YOLO8_MODELS_PATH" default="[$(find object_detector_2d)/models/breakfast_1.pt, $(find object_detector_2d)/models/breakfast_2.pt]" />
    <arg name="MODELS_PATH" default="$(find object_detector_2d)/models/gazebo/" />
    <arg name="LABELS_PATH" default="$(find object_detector_2d)/models/gazebo/label_map.pbtxt" />
    <arg name="CAMERA_FRAME" default="zed2_left_camera_optical_frame" />
    <arg name="FLIP_IMAGE" default="False" />
    <arg name="DETECTIONS_TOPIC" default="/detections" />
    <arg name="DETECTIONS_IMAGE_TOPIC" default="/detections_image" />
    <arg name="3D_MARKER_PUBLISHER" default="/detections_3d_markers" />

    <node name="Detection2D" pkg="object_detector_2d" type="MultiObjectDetector2D.py" respawn="false" output="screen">
        <param name="SOURCE" value="$(arg SOURCE)" />
        <param name="ROS_INPUT" value="$(arg ROS_INPUT)" />
        <param name="USE_ACTIVE_FLAG" value="$(arg USE_ACTIVE_FLAG)" />
        <param name="VERBOSE" value="$(arg VERBOSE)" />
        <param name="USE_YOLO8" value="$(arg USE_YOLO8)" />
        <param name="YOLO_MODEL_PATH" value="$(arg YOLO_MODEL_PATH)" />
        <param name="YOLO5_MODELS_PATH" value="$(arg YOLO5_MODELS_PATH)" />
        <param name="YOLO8_MODELS_PATH" value="$(arg YOLO8_MODELS_PATH)" />
        <param name="MODELS_PATH" value="$(arg MODELS_PATH)" />
        <param name="LABELS_PATH" value="$(arg LABELS_PATH)" />
        <param name="DEPTH_ACTIVE" value="$(arg DEPTH_ACTIVE)" />
        <param name="DEPTH_INPUT" value="$(arg DEPTH_INPUT)" />
        <param name="CAMERA_INFO" value="$(arg CAMERA_INFO)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
        <param name="FLIP_IMAGE" value="$(arg FLIP_IMAGE)" />
        <param name="DETECTIONS_TOPIC" value="$(arg DETECTIONS_TOPIC)" />
        <param name="DETECTIONS_IMAGE_TOPIC" value="$(arg DETECTIONS_IMAGE_TOPIC)" />
        <param name="3D_MARKER_PUBLISHER" value="$(arg 3D_MARKER_PUBLISHER)" />
    </node>

</launch>

